pkg/ds/daemon_set.go

```diff
@@ -392,6 +413,105 @@ func (ds *daemonSet) unschedule(node types.NodeName) error {
 	return nil
 }
 
+func (ds *daemonSet) PublishToReplication() error {
+	// TODO: We need to specifically turn this on to work
+	if ds.healthChecker == nil {
+		ds.logger.Info("Healthchecker is nil")
+		return nil
+	}
+
+	podLocations, err := ds.CurrentPods()
+	if err != nil {
+		return util.Errorf("Error retrieving pod locations from daemon set: %v", err)
+	}
+	nodes := podLocations.Nodes()
+
+	ds.logger.Infof("Preparing to publish the following nodes: %v", nodes)
+
+	thisHost, err := os.Hostname()
+	if err != nil {
+		log.Fatalf("Could not retrieve hostname: %s", err)
+	}
+	thisUser, err := user.Current()
+	if err != nil {
+		log.Fatalf("Could not retrieve user: %s", err)
+	}
+	lockMessage := fmt.Sprintf("%q from %q at %q", thisUser.Username, thisHost, time.Now())
+	repl, err := replication.NewReplicator(
+		ds.DaemonSet.Manifest,
+		ds.logger,
+		nodes,
+		len(nodes)-ds.DaemonSet.MinHealth,
+		ds.kpStore,
+		ds.applicator,
+		*ds.healthChecker,
+		health.HealthState(health.Passing),
+		lockMessage,
+	)
+	if err != nil {
+		log.Errorf("Could not initialize replicator: %s", err)
+	}
+
+	ds.logger.Info("New replicator was made")
+
+	replication, errCh, err := repl.InitializeReplication(
+		false,
+		false,
+		replication.DefaultConcurrentReality,
+	)
+	if err != nil {
+		log.Errorfs("Unable to initialize replication: %s", err)
+	}
+
+	ds.logger.Info("Replication initialized")
+
+	// auto-drain this channel
+	go func() {
+		for range errCh {
+		}
+	}()
+
+	go func() {
+		// clear lock immediately on ctrl-C
```

Comment:
This signal trap is useful in bin/p2-ds-farm but won't be useful when this is running in our server. Either: localize it to the code that's in `bin/` or remove it